community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/RecoveryVisitor.java

```diff
@@ -74,6 +75,7 @@ public boolean visit( RecoverableTransaction transaction ) throws IOException
         }
 
         lastTransactionIdApplied = txId;
+        lastTransactionCommitTimestamp = transaction.representation().getCommitEntry().getTimeWritten();
```

Comment:
Right, that should be the case as the potential commit time difference between the out of order transactions in the log should be very small in comparison to the `safeZone`.